mne/annotations.py

```diff
@@ -430,28 +430,69 @@ def _write_annotations(fid, annotations):
     end_block(fid, FIFF.FIFFB_MNE_ANNOTATIONS)
 
 
-def read_annotations(fname):
-    """Read annotations from a FIF file.
+def read_annotations(fname, sfreq='auto', uint16_codec=None):
```

Comment:
RFC: 
1. Shall we propagate `unit16_codec`? or wait for someone to raise the issue?
2. Shall we add `None` to `sfreq` and warn if it's set for a reader that does not support it?

Reply:
> Shall we propagate unit16_codec? or wait for someone to raise the issue?

What do you mean by propagate? We need it if and only if the reader is EEGLAB IIRC. So you can say in the params list that it's only used for that reader, and ignored for others.

> Shall we add None to sfreq and warn if it's set for a reader that does not support it?

In the arg list just say which file formats for which it is used (or ignored, whichever is shorter).